### DISCUSSION TITLE
Support for Hashicorp Vault / Azure Key Vault

### DISCUSSION THREAD

**Discussion by vinvli** [rabbitmq/cluster-operator#821] [Q&A] [answered]
is it possible to use a vault to retrieve secrets like erlang cookie or username/password for the default users?
Currently these secrets are stored in kubernetes itself.

**Comment by Zerpet** [comment]
We recently explored this option for Vault. [This repo](https://github.com/rabbitmq/default-user-credential-updater) is the outcome. I don't think we are currently building imges for the sidecar, if you want to try it out, you'll have to build the sidecar image and [follow this example](https://github.com/rabbitmq/cluster-operator/tree/main/docs/examples/vault-default-user).

**Comment by ChunyiLyu** [comment, marked as answer]
@vinvli not for erlang cookie but we now have Vault support for default user username/password and TLS. See: https://www.rabbitmq.com/kubernetes/operator/using-operator.html#vault
